addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/DistributionUpdateCommand.java

```diff
@@ -0,0 +1,140 @@
+package org.jboss.forge.addon.manager.impl.ui;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import net.lingala.zip4j.core.ZipFile;
+import net.lingala.zip4j.model.FileHeader;
+
+import org.jboss.forge.addon.dependencies.Coordinate;
+import org.jboss.forge.addon.dependencies.Dependency;
+import org.jboss.forge.addon.dependencies.DependencyResolver;
+import org.jboss.forge.addon.dependencies.builder.CoordinateBuilder;
+import org.jboss.forge.addon.dependencies.builder.DependencyQueryBuilder;
+import org.jboss.forge.addon.resource.FileResource;
+import org.jboss.forge.addon.ui.annotation.Command;
+import org.jboss.forge.addon.ui.annotation.predicate.NonGUIEnabledPredicate;
+import org.jboss.forge.addon.ui.context.UIExecutionContext;
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.forge.furnace.versions.SingleVersion;
+import org.jboss.forge.furnace.versions.Version;
+
+public class DistributionUpdateCommand
+{
+   @Inject
+   Addon currentAddon;
+   @Inject
+   private DependencyResolver dependencyResolver;
+   
+   // to not duplicate the code
+   private File forgeHomeDir = OperatingSystemUtils.getForgeHomeDir();
+   private File updateDir = new File(forgeHomeDir,".update");
+   
+
+   @Command(value="Forge: Update Distribution",categories={ "Forge", "Manage" }, help="Update the whole forge installation", enabled=NonGUIEnabledPredicate.class)
+   public Result updateDistribution(UIExecutionContext context) throws Exception
+   {
+      String forgeHomeDirPath = OperatingSystemUtils.getForgeHomeDir().getAbsolutePath();
+      
+      if(updateDir.exists()) {
+         return Results.fail("There is already pending update in this forge distribution."
+                  + " Restart forge to finish the update or "
+                  + "run update-distribution-cancel command to cancel the updating process");
+      }
+      if (!context.getPrompt().promptBoolean(
```

Comment:
The prompt should happen only if a version is found. It's pointless to prompt the user if nothing is going to happen